### PR TITLE
Annotations and labels

### PR DESCRIPTION
**Pull request by gregorjerse**
Search and filter by label when vocabulary is present.

### CODE REVIEW

File: resolwe/flow/tests/test_annotations.py [hunk 1]
```diff
@@ -511,35 +554,32 @@ def test_annotation_field(self):
...
         self.assertEqual(response.data[0]["name"], "field1")
 
         request = factory.get("/", {"label": "Annotation field 1"}, format="json")
-        force_authenticate(request, self.contributor)
         response = self.annotationfield_viewset(request)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 1)
         self.assertEqual(response.data[0]["name"], "field1")
 
         request = factory.get("/", {"label__icontains": "fiELd 1"}, format="json")
```
Comment (acopar) [inline resolwe/flow/tests/test_annotations.py:582]: Is this a typo? Probably `field 1`?

File: resolwe/flow/views/entity.py [hunk 2]
```diff
@@ -73,9 +73,10 @@ def order_queryset(self, queryset):
             fields = [field.strip() for field in ordering.split(",")]
             for match in filter(None, map(regex.match, fields)):
                 field_id = match.group("field_id")
+                # Always filter by user-visible attribute label.
                 annotation_value = AnnotationValue.objects.filter(
```
Comment (dblenkus) [inline resolwe/flow/views/entity.py:77]: You should rename this to `annotation_label`.

Comment (gregorjerse) [reply resolwe/flow/views/entity.py:77]: Done.

File: resolwe/flow/views/annotations.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ class AnnotationValueViewSet(
 ):
     """Annotation value viewset."""
 
-    permission_classes = (permissions.IsAuthenticated,)
+    permission_classes = (permissions.AllowAny,)
```
Comment (dblenkus) [inline resolwe/flow/views/annotations.py:71]: Add a comment.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**